ARROW-16311: [Java] Do not return table_schema column when it's not requested

timvw opened the pull request:
Currently, when the FlightInfo for CommandGetTables is requested, it always (hardcoded) returns a schema which includes the table_schema column (Schema.GET_TABLES_SCHEMA)  even when the table_schema is not requested (include_schema on CommandGetTables = false).

lidavidm reviewed (commented):
Thanks for catching this.

Can we add a test here? https://github.com/apache/arrow/blob/7eb60eff53fde5a24f90d3d87f33737882819ef0/java/flight/flight-sql/src/test/java/org/apache/arrow/flight/TestFlightSql.java#L177-L181

There's also a style/lint warning:

```
Warning:  src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java:[1495,5] (whitespace) WhitespaceAround: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
```

timvw commented:
* Added a test to assert on the schema when include_schema is set to false
* Changed implementation to leverage existing Schemas.GET_TABLES_SCHEMA_NO_SCHEMA
* Fixed issue with whitespace after "if" statement

lidavidm commented on java/flight/flight-sql/src/test/java/org/apache/arrow/flight/TestFlightSql.java:
```diff
@@ -180,6 +180,14 @@ public void testGetTablesSchema() {
   }
 
+  @Test
+  public void testGetTablesSchemaExcludeSchema() {
+    final FlightInfo info = sqlClient.getTables(null, null, null, null, false);
+    collector.checkThat(info.getSchema(), is(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA_NO_SCHEMA));
+  }
+
+
+
```
nit: no need for so many newlines
```suggestion
  public void testGetTablesSchemaExcludeSchema() {
    final FlightInfo info = sqlClient.getTables(null, null, null, null, false);
    collector.checkThat(info.getSchema(), is(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA_NO_SCHEMA));
  }

```

lidavidm commented on java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java:
```diff
@@ -1488,7 +1488,14 @@ public void getStreamSchemas(final CommandGetDbSchemas command, final CallContex
   public FlightInfo getFlightInfoTables(final CommandGetTables request, final CallContext context,
                                         final FlightDescriptor descriptor) {
-    return getFlightInfoForSchema(request, descriptor, Schemas.GET_TABLES_SCHEMA);
+
+    Schema schemaToUse = Schemas.GET_TABLES_SCHEMA;
+
+    if (!request.getIncludeSchema()) {
+      schemaToUse = Schemas.GET_TABLES_SCHEMA_NO_SCHEMA;
+    }
+
+    return getFlightInfoForSchema(request, descriptor, schemaToUse);
```
nit: no need for so many newlines
```suggestion
    Schema schemaToUse = Schemas.GET_TABLES_SCHEMA;
    if (!request.getIncludeSchema()) {
      schemaToUse = Schemas.GET_TABLES_SCHEMA_NO_SCHEMA;
    }

    return getFlightInfoForSchema(request, descriptor, schemaToUse);
```

lidavidm reviewed (approved):
Thanks!
